### PR TITLE
using trie enhance dubbo route.go

### PR DESCRIPTION
**Pull request by yqxu**
using trie enhance dubbo route.go

### CODE REVIEW

**Comment by ztelur** [comment]
plz fix ci problem

File: pkg/common/router/trie/trie.go [hunk 1]
```diff
@@ -166,6 +172,17 @@ func (node *Node) internalPut(keys []string, bizInfo interface{}) (bool, error)
 
 }
 
+func (node *Node) Clear() bool {
```
Comment (AlexStocks) [inline pkg/common/router/trie/trie.go:175]: 真是迂腐。'*node = Node{}' 一行代码 enough

File: pkg/common/router/trie/trie_test.go [hunk 2]
```diff
@@ -18,7 +18,10 @@
 package trie
 
 import (
+	"github.com/apache/dubbo-go-pixiu/pkg/common/util/stringutil"
```
Comment (AlexStocks) [inline pkg/common/router/trie/trie_test.go]: 这...

File: pkg/router/route_test.go [hunk 3]
```diff
@@ -99,72 +92,73 @@ func TestFindMethod(t *testing.T) {
...
-	assert.False(t, ok)
-	_, ok = rt.searchWildcard("/theboys/:id/age")
-	assert.False(t, ok)
-	_, ok = rt.searchWildcard("/theboys/butcher")
-	assert.True(t, ok)
-	_, ok = rt.searchWildcard("/vought/:id/supe/homelander")
-	assert.True(t, ok)
-}
+//
+//func TestUpdateMethod(t *testing.T) {
```
Comment (AlexStocks) [inline pkg/router/route_test.go:116]: 确定无用了吗？无用就删掉。

File: pkg/router/route.go [hunk 4]
```diff
@@ -26,131 +26,117 @@ import (
...
-		for _, v := range splitedURLs {
-			if v != api.IntegrationRequest.HTTPBackendConfig.URL {
-				afterRemoveedURL = append(afterRemoveedURL, v)
-			}
-		}
-		if len(afterRemoveedURL) == 0 {
-			delete(node.methods, api.HTTPVerb)
+	lowerCasePath := strings.ToLower(api.URLPattern)
+	key := getTrieKey(api.Method.HTTPVerb, lowerCasePath, false)
+	_, _ = r.tree.Remove(key)
```
Comment (PhilYue) [inline pkg/router/route.go:65]: 这里 remove 操作是否要加锁🔐 ？

Comment (yqxu) [reply pkg/router/route.go:65]: thx 需要的

File: pkg/common/util/stringutil/stringutil.go [hunk 5]
```diff
@@ -72,14 +72,19 @@ func IsMatchAll(key string) bool {
...
 	if strings.HasPrefix(path, constant.PathSlash) {
 		ret = method + path
 	} else {
 		ret = method + constant.PathSlash + path
 	}
-
 	if strings.HasSuffix(ret, constant.PathSlash) {
 		ret = ret[0 : len(ret)-1]
 	}
+	ret = strings.Split(ret, "?")[0]
```
Comment (ztelur) [inline pkg/common/util/stringutil/stringutil.go:88]: is it better to make "?" as a constant named variable? code in trie.go uses "?" too

File: pkg/common/router/trie/trie.go [hunk 6]
```diff
@@ -18,12 +18,13 @@
...
 
 import (
-	"github.com/pkg/errors"
+	"strings"
 )
 
 import (
 	"github.com/apache/dubbo-go-pixiu/pkg/common/util/stringutil"
 	"github.com/apache/dubbo-go-pixiu/pkg/logger"
+	"github.com/pkg/errors"
```
Comment (ztelur) [inline pkg/common/router/trie/trie.go:27]: please split third party import to another import block,like below.

import (
	"github.com/apache/dubbo-go-pixiu/pkg/common/util/stringutil"
	"github.com/apache/dubbo-go-pixiu/pkg/logger"
)

import (
       "github.com/pkg/errors""
)

**Review by ztelur — APPROVED** [review]

**Review by PhilYue — APPROVED** [review]

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**